FirebaseRemoteConfig/SwiftNew/RemoteConfig.swift

```diff
@@ -0,0 +1,942 @@
+
+  private let configRealtime: ConfigRealtime
+
+  private let queue: DispatchQueue
+
+  // TODO: remove objc public/
+  @objc public let appName: String
+
+  private var listeners = [RemoteConfigListener]()
+
+  public var FIRNamespace: String
+
+  /// Shared Remote Config instances, keyed by FIRApp name and namespace.
+  private static var RCInstances = [String: [String: RemoteConfig]]()
+
+  // MARK: - Public Initializers and Accessors
+
+  @objc public static func remoteConfig(with app: FirebaseApp) -> RemoteConfig {
+    return remoteConfig(withFIRNamespace: RemoteConfigConstants.NamespaceGoogleMobilePlatform,
+                        app: app)
+  }
+
+  @objc public static func remoteConfig() -> RemoteConfig {
+    guard let app = FirebaseApp.app() else {
+      fatalError("The default FirebaseApp instance must be configured before the " +
+        "default Remote Config instance can be initialized. One way to ensure " +
+        "this is to call `FirebaseApp.configure()` in the App Delegate's " +
+        "`application(_:didFinishLaunchingWithOptions:)` or the `@main` struct's " +
+        "initializer in SwiftUI.")
+    }
+    return remoteConfig(withFIRNamespace: RemoteConfigConstants.NamespaceGoogleMobilePlatform,
+                        app: app)
+  }
+
+  @objc(remoteConfigWithFIRNamespace:)
+  public static func remoteConfig(withFIRNamespace firebaseNamespace: String) -> RemoteConfig {
+    guard let app = FirebaseApp.app() else {
+      fatalError("The default FirebaseApp instance must be configured before the " +
+        "default Remote Config instance can be initialized. One way to ensure " +
+        "this is to call `FirebaseApp.configure()` in the App Delegate's " +
+        "`application(_:didFinishLaunchingWithOptions:)` or the `@main` struct's " +
+        "initializer in SwiftUI.")
+    }
+
+    return remoteConfig(withFIRNamespace: firebaseNamespace, app: app)
+  }
+
+  // Use the provider to generate and return instances of FIRRemoteConfig for this specific app and
+  // namespace. This will ensure the app is configured before Remote Config can return an instance.
+  @objc(remoteConfigWithFIRNamespace:app:)
+  public static func remoteConfig(withFIRNamespace firebaseNamespace: String = RemoteConfigConstants
+    .NamespaceGoogleMobilePlatform,
+    app: FirebaseApp) -> RemoteConfig {
+    let provider = ComponentType<RemoteConfigInterop>
+      .instance(
+        for: RemoteConfigInterop.self,
+        in: app.container
+      ) as! any RemoteConfigProvider as RemoteConfigProvider
+    return provider.remoteConfig(forNamespace: firebaseNamespace)!
+  }
+
+  /// Last successful fetch completion time.
+  @objc public var lastFetchTime: Date? {
+    var fetchTime: Date?
+    queue.sync {
+      let lastFetchTimeInterval = self.settings.lastFetchTimeInterval
+      if lastFetchTimeInterval > 0 {
+        fetchTime = Date(timeIntervalSince1970: lastFetchTimeInterval)
+      }
```

Comment:
I think this is ok as-is, but the `if` block is exclusive to the Swift implementation.

```objc
- (NSDate *)lastFetchTime {
  __block NSDate *fetchTime;
  dispatch_sync(_queue, ^{
    NSTimeInterval lastFetchTime = self->_settings.lastFetchTimeInterval;
    fetchTime = [NSDate dateWithTimeIntervalSince1970:lastFetchTime];
  });
  return fetchTime;
}
```

Reply:
made consistent